packages/core/src/agent/MessageSender.ts

```diff
@@ -48,15 +48,15 @@ export class MessageSender {
   public constructor(
     envelopeService: EnvelopeService,
     transportService: TransportService,
-    @inject(InjectionSymbols.MessageRepository) messageRepository: MessageRepository,
+    @inject(InjectionSymbols.MessagePickupRepository) messagePickupRepository: MessagePickupRepository,
```

Comment:
At the beginning I named it `MessagePickupQueue` but discussing about it with my colleagues we noticed that we are not using it as a _queue_, in the sense that we are enqueuing data to it but not exactly dequeuing data when retrieving it (as we must store until it is actually acknowledged by the mediatee). Also, the data on it is somewhat organized (i.e. by connectionId, recipientKey, state, etc.) so the term 'Repository' feels to me more appropriate than 'Storage', which is usually used to generically save data.

Having said this, I'm not happy because the unfortunate name clash with `Repository` interface, so I'm more than open to opinions about how to improve the naming!

Reply:
I think `MessageRepository` is best i.e. it's _the place where you store and fetch messages_. I think `xxxQueue` would tend to incorrectly enforce the underlying technology, when any persistent storage may actually work. `Pickup` is just a fragment of the interface (we're not 'picking up' messages received that cannot be delivered, rather we're queuing them for later pickup), so I would just drop it to avoid narrowing the exhibited responsibility.

Reply:
The problem with `MessageRepository` is that it collides with `Repository` classes used within the framework, most notably `DidCommMessageRepository`, so leaving such generic name would be more confusing. The 'Pickup' word tries to specify that it is used for message pickup protocol.

It seems it is still not clear, as this `MessagePickupRepository` is an interface rather than an implementation, like the other `Repository`.